compute/metadata/metadata.go

```diff
@@ -408,17 +412,41 @@ func strsContains(ss []string, s string) bool {
 
 // A Client provides metadata.
 type Client struct {
-	hc *http.Client
+	hc     *http.Client
+	logger *slog.Logger
+}
+
+// Options for configuring a [Client].
+type Options struct {
+	// Client is the HTTP used to make requests. Optional.
+	Client *http.Client
+	// Logger is used to log information about outgoing requests. Optional.
```

Comment:
"to log information about outgoing requests" isn't accurate, because responses are also logged. Maybe: "to log HTTP requests and responses".

It would probably also be good to disclose what is logged: "The structured log statements for requests will include method, URL, headers and payload. The structured log statements for responses will include status, headers and payload. 

And we should add a reassurance here about the default behavior. Maybe: "If not provided, requests and responses will not be logged."

Reply:
fixed up a bit. Did not give low level details of what is logged or the format as this is an impl details that can change over time and/or by the logger that is passed in.